### PR TITLE
CLDR-16643 Fix kh, add more comprehensive tests

### PR DESCRIPTION
**Pull request by macchiati**
CLDR-16643

Most of the work was in fleshing out the test to try to prevent breakage before it hits ICU. Didn't have time to port the ICU tests, so just did a quick test.

- [ ] This PR completes the ticket.

ALLOW_MANY_COMMITS=true

### CODE REVIEW

**Review by pedberg-icu — APPROVED** [review]
Nice to have more transform testing on the CLDR side!

**Comment by pedberg-icu** [comment]
Hmm. The ticket is for 43.1 but this PR is against the main branch. Guess I will cherry-pick it to maint/maint-43 branch (easier with this commit than with the spec one) when I cherry pick the relevant maint/maint-43 fixes in the other direction, to main.

**Review by srl295 — APPROVED** [review]

File: common/transforms/Cyrillic-Latin.xml [hunk 1]
```diff
@@ -33,7 +33,12 @@ $descender = ˌ;
 $umlaut = \u0308 ;
 $macron = \u0304 ;
 $breveBelow = \u032E;
```
Comment (markusicu) [inline common/transforms/Cyrillic-Latin.xml:35]: Is this unused now?

**Review by markusicu — COMMENTED** [review]
thanks! one question...

File: common/transforms/Cyrillic-Latin.xml (hunk 1 above)
Comment (macchiati) [reply common/transforms/Cyrillic-Latin.xml:35]: Just checked, and it isn't used. (I don't think this is worth a new PR; we haven't been concerned about details like that in the files.

**Pull request merged**